src/models/hooks/activityReport.js

```diff
@@ -324,17 +324,52 @@ const propagateApprovedStatus = async (sequelize, instance, options) => {
       }
 
       if (objectives && objectives.length > 0) {
```

Comment:
A comment like 'when the report is approved mark all related objective models as onApproved' true.